google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -2081,6 +2081,51 @@ def self.gapi_from_attrs gapi, attributes
           Google::Apis::StorageV1::Object.new(**attr_params)
         end
 
+        ##
+        # from_gs_url is a method to fetch bucket details and file details from a gs url
+        #
+        # @return [Hash(String => String)]
+        #
+        # @example Fetch bucket_name and file_name from gs url:
+        #   require "google/cloud/storage"
+        #   gs_url= "gs://my-todo-app/avatars/heidi.jpeg"
+        #   file=Google::Cloud::Storage::File
+        #   file.from_gs_url(gs_url)
+        #   =>
+        #   {"bucket_name"=>"my-todo-app", "file_name"=>"heidi.jpeg"}
+        #
+        # @example Fetch bucket_name , file_name and other query params from gs url:
+        #   require "google/cloud/storage"
+        #   gs_url= "gs://my-todo-app/test_sub_folder/heidi.jpeg?params1=test1&params2=test2"
+        #   file=Google::Cloud::Storage::File
+        #   file.from_gs_url(gs_url)
+        #   =>{
+        #     "bucket_name"=>"my-todo-app",
+        #     "file_name"=>"heidi.jpeg",
+        #     "options" => {
+        #       "params1"=>"test1",
+        #       "params2"=>"test2"
+        #       }
+        #      }
+
+        def self.from_gs_url gs_url
+          prefix = "gs://".freeze
+          raise ArgumentError, "Invalid GCS URL" unless gs_url.start_with? prefix
+          # seprating params from input url
+          path, query = gs_url.sub(prefix, "").split("?", 2)
+          # parsing the url
+          bucket_name, file_path = path.split "/", 2
+          query_params = URI.decode_www_form(query).to_h if query
+          file_name = file_path.split("/").last
```

Comment:
I guess we can leave the `file_path` as is.

Reply:
so we don't need to extract the file name from path correct?
```
file_name = file_path
```

Reply:
Just returning the file_path should be good.